lms/views/admin/application_instance.py

```diff
@@ -180,6 +180,9 @@ def upgrade_instance(self, lti_registration, consumer_key):
             # Flush here to find if we are making a duplicate in the process of upgrading
             self.request.db.flush()
         except IntegrityError:
+            # Leave a clean transaction, otherwise  we get a:
+            #   "This Session's transaction has been rolled back due to a previous exception during flush."
+            self.request.db.rollback()
```

Comment:
Not sure how to assert this call in the tests. 

Mocking request.db for this one test?

Reply:
You could purposefully break the DB handle somehow?

Reply:
Turns out this is difficult, but we can check that the session is still in working order after. I've made this change and will release this as Marcos is away.

I figure having two of us co-operate kind of makes up a review?